samples/bluetooth/rssi_power_control/central/src/main.c

```diff
@@ -0,0 +1,275 @@
+		BT_CONN_LE_OPT_NONE,
+		BT_GAP_SCAN_FAST_INTERVAL,
+		BT_GAP_SCAN_FAST_INTERVAL
+	);
+
+	err = bt_conn_le_create(device_info->recv_info->addr, conn_params,
+				BT_LE_CONN_PARAM_DEFAULT, &default_conn);
+	if (err) {
+		LOG_ERR("Connection failed (err: %d)", err);
+		bt_scan_start(BT_SCAN_TYPE_SCAN_ACTIVE);
+		return;
+	}
+
+	LOG_INF("Connection to %s", addr);
+}
+
+/* Initialize a scan callback structure */
+BT_SCAN_CB_INIT(scan_cb, scan_filter_match, NULL, NULL, NULL);
+
+/* Function to initialize automated power control */
+static void init_power_control(void)
+{
+	sdc_hci_cmd_vs_set_power_control_request_params_t p_params = {
+		.beta = POWER_CONTROL_SENSITIVITY,
+		.auto_enable = ENABLED,
+		.lower_limit = LOWER_LIMIT_RSSI,
+		.upper_limit = UPPER_LIMIT_RSSI,
+		.lower_target_rssi = LOWER_TARGET_RSSI,
+		.upper_target_rssi = UPPER_TARGET_RSSI,
+		.wait_period_ms = WAIT_TIME_MS
+	};
+
+	int err = sdc_hci_cmd_vs_set_power_control_request_params(&p_params);
+
+	if (err) {
+		LOG_ERR("Failed to initialize power control");
+		return;
+	}
+	LOG_INF("Successful initialization of power control");
+}
+
+/* Initializes the scanning process */
+static void scan_init(void)
+{
+	/* Scan parameters with fast scanning intervals and windows */
+	struct bt_le_scan_param scan_param = {
+			.type = BT_LE_SCAN_TYPE_ACTIVE,
+			.interval = BT_GAP_SCAN_FAST_INTERVAL,
+			.window = BT_GAP_SCAN_FAST_WINDOW,
+	};
+
+	/* Struct to initialize the scan module */
+	struct bt_scan_init_param scan_init = {
+		/* Disable automatic connection. Instead handle
+		 * connection manually through scan_filter_match()
+		 */
+		.connect_if_match = DISABLED,
+		/* Links to a bt_le_scan_param structure, defined earlier */
+		.scan_param = &scan_param,
+	};
+
+
+	bt_scan_init(&scan_init);
+	bt_scan_cb_register(&scan_cb);
+
+	int err = bt_scan_filter_add(BT_SCAN_FILTER_TYPE_NAME, PERIPHERAL_NAME);
+
+	if (err) {
+		LOG_ERR("Failed to add peripheral name filter (err %d)", err);
+		return;
+	}
+
+	err = bt_scan_filter_enable(BT_SCAN_NAME_FILTER, false);
+	if (err) {
+		LOG_ERR("Failed to enable filter (err: %d)", err);
+		return;
+	}
+}
+
+/* Functionality to read the RSSI value */
+static void read_conn_rssi(uint16_t handle, int8_t *rssi)
+{
+	struct net_buf *buf, *rsp = NULL;
+	struct bt_hci_cp_read_rssi *cp;
+	struct bt_hci_rp_read_rssi *rp;
+
+	int err;
+
+
+	buf = bt_hci_cmd_alloc(K_FOREVER);
+	if (!buf) {
+		LOG_ERR("Unnable to allocate command buffer");
+		return;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	cp->handle = sys_cpu_to_le16(handle);
+
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_READ_RSSI, buf, &rsp);
+	if (err) {
+		LOG_ERR("Read RSSI err: %d", err);
+		return;
+	}
+
+	rp = (void *)rsp->data;
+	*rssi = rp->rssi;
+
+	net_buf_unref(rsp);
+}
+
+/* Work handler to poll Tx and RSSI periodically */
+static void tx_rssi_work_handler(struct k_work *work)
+{
+	if (!default_conn) {
+		LOG_ERR("No connection, do not reschedule");
+		return;
+	}
+
+	uint16_t conn_handle;
```

Comment:
declare all variables at top of scope, not half way through a function (fine to declare at the top of an `if() { ...` if they are used there but simply not like this, fix in whole PR